### ISSUE TITLE
Compare cached JSON with JSON on server

### ISSUE THREAD

**Issue by thomascsorbamedia** [Haneke/HanekeSwift#159]
I want to know if it is possible to compare the json i cache to the json on the server?
I have a json query that may change, if i change it the query is different, when i go to the app (killed it first) i expect to see the different results. The problem is that i can only see the cached data.

Possible other way:
Maybe reverse the order of checking, always check with source and if source is different then refetch all data?

quote from readme
"memory, disk or NSURLCache. If not available, Haneke will fetch the JSON from the source"

If there someone can help me asap i would be very thankful, because i'm working on a project with a deadline :)

Greets, Thomas

**Comment by thomascsorbamedia** [comment]
There i nobody going to answer this? it seems to me that it is kind of a requested feature to update every time and append to existing while having a internet connection...

**Comment by thomascsorbamedia** [comment]
at this moment my only solutions seems to be to delete and redownload the json overtime the device has an internet connection, there is one problem though the images that are downloaded and later on don't exist in the json anymore are not removed, they stay in documents folder.

Main problem is that the json cannot be updated automatically *feature request?*

**Comment by basememara** [comment]
Maybe you can add a version # to the JSON? I'm facing this same issue and in the past with JavaScript, I added a web service that just returns the lastModified date of the API request. If it's newer than the cached date, then I perform an update to the cache. Still trying to figure how to implement this here. I agree, it would be a handy feature though, such as an optional parameter:

`let cache = HNKCache<JSON>(name: "dasboard", versionURL: "http://example.com/api/products/version")`

Then internally the framework can call the versionUrl before deciding whether to make the full JSON request. Only catch is that there will ALWAYS be an async call involved even if cached, but a least the versionURL should be much more responsive than the full JSON request.

**Comment by basememara** [comment]
Another way may be to add an optional expiry interval:
`let cache = HNKCache<JSON>(name: "dasboard", expiry: NSTimeInterval(86400))`

**Comment by tobiasoleary** [comment]
Are you saying that you would like Haneke to check Etags?  Or are you saying you would like Haneke to do the server request each time then compare the results with the cached version of the results?

**Comment by basememara** [comment]
Originally I was talking about doing a partial server request just to get the version #.. but Etags is a great idea!!!

**Comment by tobiasoleary** [comment]
@basememara Yeah. I meant to pose those question to the original creator of the issue. I haven't done a deep dive into Haneke, but I believe by default it uses `NSURLSession` for requests, which should automatically support entity tags if the `NSURLSession` has a configuration with its `requestCachePolicy` set to  `NSURLCacheStoragePolicy.UseProtocolCachePolicy`.

**Comment by basememara** [comment]
There's an elaborate question/answer on StackOverflow that can be used as a reference here, it mentions Haneke as well: http://stackoverflow.com/questions/28418035/synchronizing-remote-json-data-to-local-cache-storage-in-ios-swift

**Comment by tobiasoleary** [comment]
@basememera Wow. That was a great question and answer. Thank you for sharing.
